#!/usr/bin/env python

from twython import Twython, TwythonError
from redis import Redis
import os


redis_url = os.getenv('REDISTOGO_URL', 'redis://localhost:6379')
r = Redis.from_url(redis_url)

APP_KEY = "kHLxa38Nt8sT8P9lz4ymRcgHT"
APP_SECRET = "MLZzxOoFCBCtOVCkN1yssGrEMGiR7U4BW4oGoxRphvv0zoUlWp"

token = r.get("twitter:token")
secret = r.get("twitter:secret")
twitter = Twython(APP_KEY, APP_SECRET, token, secret)

with open("keywords.txt") as fc:
  number_of_keywords = len(fc.readlines())

#follow by keywords 
with open("keywords.txt") as f:
  per_keyword = int(1000/number_of_keywords/24)
  for line in f:
    keyword = line.strip()
    followed = 0
    for i in range(10):
        people = twitter.search_users(q=keyword, page=i)
        print(len(people))
        for p in people:
            print(per_keyword)
            print(followed)
            if followed <= per_keyword:
              try:
                twitter.create_friendship(user_id=p['id'])
                r.hset(p['id'], 'follows', False)
                r.hset(p['id'], 'keyword', keyword)
                followed += 1;
              except TwythonError as e:
                pass
            else:
              break

